server/src/services/deleteReview.js

```diff
@@ -0,0 +1,13 @@
+import { Review, Vote } from "../models/index.js"
+
+const deleteReview = async (reviewId) => {
+  const review = await Review.query().findById(reviewId)
+  const votes = await review.$relatedQuery("votes")
+  for (const vote of votes) {
+    await Vote.query().deleteById(vote.id)
+  }
+  await Review.query().deleteById(reviewId)
+  return true
+}
```

Comment:
while this is a thoughtful approach, it's a lot of overhead for us to manage as we add new associations to the application overtime! Instead we can rely on the database to handle this for us!

in the migration for the "child" entity in the association (the table that defines the foreign key), add the chainable method `onDelete("CASCADE")` which says "when the parent resource is destroyed, also delete these dependent children"

https://knexjs.org/guide/schema-builder.html#ondelete

so add that chainable method to the reviewId in the votes table - you may consider this approach for other resources that are deletable (where applicable in your app)